### ISSUE TITLE
Questions about how DDP handles gradients

### ISSUE THREAD

**Issue by helson73** [NVIDIA/apex#696]
I have two questions about how DDP works. 
1) When multi-GPU is used with apex DDP, how gradients from each GPU summarized? Are they averaged? If so, is it equivalent to loss function scaled with : loss = sum(loss) / batch_size / world_size?
2) In imagenet example, learning rate is scaled by factor batch_size * world_size / 256, is it equivalent to loss = sum(loss) / 256 ?

**Comment by krishansubudhi** [comment]
> Parameters are broadcast across participating processes on initialization, and gradients are allreduced and averaged over processes during backward().

No need to divide by world_size.
[Source](https://nvidia.github.io/apex/parallel.html#apex.parallel.DistributedDataParallel)

**Comment by helson73** [comment]
@krishansubudhi  
Yeah, but somehow the average loss is incorrect.
